### PR TITLE
fix(fetch/server) fix server end of stream, fix fetch not streaming without content-length or chunked encoding, fix case when stream do not return a promise on pull

### PR DESCRIPTION
**Pull request by cirospaciari**
…gth or chunked encoding, fix case when stream do not return a promise on pull

### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests + Existing Tests

### CODE REVIEW

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -2255,7 +2267,12 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
 
             this.setAbortHandler();
             streamLog("is in progress, but did not return a Promise. Finalizing request context", .{});
-            this.finalize();
+            response_stream.sink.onFirstWrite = null;
+            response_stream.sink.ctx = null;
+            response_stream.detach();
+            stream.cancel(globalThis);
```
Comment (Jarred-Sumner) [inline src/bun.js/api/server.zig:2273]: Does `detach` call cancel?

File: test/js/web/fetch/body-stream.test.ts [hunk 2]
```diff
@@ -13,6 +13,35 @@ var port = 0;
   ];
   const useRequestObjectValues = [true, false];
 
+  test("Should not crash when not returning a promise when stream is in progress", async () => {
```
Comment (Jarred-Sumner) [inline test/js/web/fetch/body-stream.test.ts:16]: I feel like we need more tests

File: src/bun.js/api/server.zig [hunk 3]
```diff
@@ -2223,6 +2226,14 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
                         },
                     }
                     return;
+                } else {
+                    // if is not a promise we treat it as Error
+                    streamLog("returned an error", .{});
+                    if (!this.flags.aborted) resp.clearAborted();
+                    response_stream.detach();
+                    this.sink = null;
+                    response_stream.sink.destroy();
```
Comment (Jarred-Sumner) [inline src/bun.js/api/server.zig:2235]: I don't feel great about this code
